packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenuItemWrapper/ContextualMenuAnchor.tsx

```diff
@@ -0,0 +1,136 @@
+  private _anchor = createRef<HTMLAnchorElement>();
+
+  public render() {
+    const {
+      item,
+      classNames,
+      index,
+      focusableElementIndex,
+      totalItemCount,
+      hasCheckmarks,
+      hasIcons,
+      contextualMenuItemAs: ChildrenRenderer = ContextualMenuItem,
+      expandedMenuItemKey,
+      onItemClick,
+      openSubMenu,
+      dismissSubMenu,
+      dismissMenu
+    } = this.props;
+
+    let anchorRel = item.rel;
+    if (item.target && item.target.toLowerCase() === '_blank') {
+      anchorRel = anchorRel ? anchorRel : 'nofollow noopener noreferrer';  // Safe default to prevent tabjacking
+    }
+
+    const subMenuId = this._getSubMenuId(item);
+    const itemHasSubmenu = hasSubmenu(item);
+    const nativeProps = getNativeProps(item, anchorProperties);
+    const disabled = isItemDisabled(item);
+
+    let { keytipProps } = item;
+    if (keytipProps && itemHasSubmenu) {
+      keytipProps = {
+        ...keytipProps,
+        hasMenu: true
+      };
+    }
+
+    return (
+      <div>
+        <KeytipData
+          keytipProps={ item.keytipProps }
+          ariaDescribedBy={ (nativeProps as any)['aria-describedby'] }
+          disabled={ disabled }
+        >
+          { (keytipAttributes: any): JSX.Element => (
+            <a
+              { ...nativeProps }
+              { ...keytipAttributes }
+              ref={ this._anchor }
+              href={ item.href }
+              target={ item.target }
+              rel={ anchorRel }
+              className={ classNames.root }
+              role='menuitem'
+              aria-owns={ item.key === expandedMenuItemKey ? subMenuId : undefined }
+              aria-haspopup={ itemHasSubmenu || undefined }
+              aria-expanded={ itemHasSubmenu ? item.key === expandedMenuItemKey : undefined }
+              aria-posinset={ focusableElementIndex + 1 }
+              aria-setsize={ totalItemCount }
+              aria-disabled={ isItemDisabled(item) }
+              style={ item.style }
+              onClick={ this._onItemClick }
+              onMouseEnter={ this._onItemMouseEnter }
+              onMouseLeave={ this._onItemMouseLeave }
+              onKeyDown={ itemHasSubmenu ? this._onItemKeyDown : null }
+            >
+              <ChildrenRenderer
+                componentRef={ item.componentRef }
+                item={ item }
+                classNames={ classNames }
+                index={ index }
+                onCheckmarkClick={ hasCheckmarks && onItemClick ? onItemClick.bind(this, item) : undefined }
+                hasIcons={ hasIcons }
+                openSubMenu={ openSubMenu }
+                dismissSubMenu={ dismissSubMenu }
+                dismissMenu={ dismissMenu }
+                getContainerElement={ this._getContainerElement }
+              />
+            </a>
+          ) }
+        </KeytipData>
+      </div>);
+  }
+
+  private _onItemMouseEnter = (ev: React.MouseEvent<HTMLElement>): void => {
```

Comment:
Since these are the same as the ButtonItem methods, they should either get passed in as props, or inherited from some base component to reduce code duplication.